semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/ToplevelClassfile.scala

```diff
@@ -1,9 +1,18 @@
 package scala.meta.internal.metacp
 
+import scala.meta.cli.Reporter
 import scala.meta.internal.io._
 import scala.meta.io._
+import scala.meta.metacp.Settings
 import scala.tools.asm.tree._
 
-final case class ToplevelClassfile(base: AbsolutePath, path: AbsolutePath, node: ClassNode) {
+final case class ToplevelClassfile(
+    base: AbsolutePath,
+    path: AbsolutePath,
+    node: ClassNode,
+    lookup: ClasspathLookup,
+    settings: Settings,
+    reporter: Reporter
+) {
```

Comment:
I believe that `lookup`, `settings` and `reporter` should be separate from `ToplevelClassfile` and be passed as parameters to `Scalacp` and `Javacp`. This is somewhat more hassle than the current arrangement, but this will prevent us from ending up with a Global/Context design, where we have a single data structure that is a grab bag of stuff from a bunch of different submodules.

Reply:
No wrapper class, since these entities are unrelated. Just three independent parameters, in addition to `classfile`.

Reply:
Sounds good 👍 much better!